src/services/trial/useStartTrial.ts

```diff
@@ -75,16 +78,8 @@ export const useStartTrial = () => {
       queryClient.invalidateQueries(['accountDetails'])
       queryClient.invalidateQueries(['GetPlanData'])
       queryClient.invalidateQueries(['GetAvailablePlans'])
-
-      renderToast({
-        type: 'generic',
-        title: '14 day trial has started',
-        content: '',
-        options: {
-          duration: 5000,
-          position: 'bottom-left',
-        },
-      })
+      localStorage.setItem(LOCAL_STORAGE_USER_STARTED_TRIAL_KEY, 'true')
```

Comment:
Rather than storing this in `localStorage` what do you think about just adding a query param to the redirect, something like `?trial_started=true`, that way we should be able to avoid using a useEffect, which we should be doing at all costs.

Reply:
I did this to avoid making changes in `App.jsx` around handling redirects. I'm not sure if we can avoid `useEffect` in that scenario either if there are multiple re-renders so both solutions could work.

Reply:
What would you need to change in `App` if you were to use a query param?

Reply:
Adding some additional logic to `HomePageRedirect` + tests and all of that

Reply:
Yea homepage redirect is not setup at all to handle multiple query params, we should take a peak at fixing that.